### PR TITLE
Httpxe specification tests

### PR DESCRIPTION
**Pull request by jitsni**

### CODE REVIEW

**Comment by jitsni** [comment]
Needs k3po release (so do NOT merge yet)

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpRequestDecodingState.java [hunk 1]
```diff
@@ -138,7 +138,10 @@ protected DecodingState finishDecode(List<Object> childProducts,
                 hostHeaderValues.add(requestURI.getHost()
                         + (requestURI.getPort() == -1 ? "" : ":" + requestURI.getPort()));
                 headers.put(HEADER_HOST, hostHeaderValues);
-                requestURI = URI.create(requestURI.getPath());
+				String query = requestURI.getQuery();
```
Comment (cmebarrow) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpRequestDecodingState.java]: Please replace any tab characters with spaces

**Comment by cmebarrow** [comment]
@jitsni Jitu, is this ready to merge yet?

**Comment by jitsni** [comment]
I will close this one and open a new PR. Somehow this PR doesn't show the correct diff.

**Comment by jitsni** [comment]
Created a new PR https://github.com/kaazing/gateway/pull/413

**Pull request closed without merging**